api/specs/webserver/openapi-publications.yaml

```diff
@@ -0,0 +1,35 @@
+paths:
+  /publications/service-submission:
+    post:
+      tags:
+        - publication
+      summary: Submits a new service candidate
+      operationId: service_submission
+      requestBody:
+        content:
+          multipart/form-data:
+            schema:
+              type: object
+              required:
+                - data
+              properties:
+                data:
+                  type: string
+                  format: binary
+                file:
+                  type: string
+                  format: binary
```

Comment:
what is the difference between data and file? why both are required?

Reply:
Data contains the json with provided information (text) and file contains the file. I did it like this because to send a file efficiently I have to use multipart. The type of the data will always be ``application/json`` but the type of the file can vary.
The other option was to encode the file contents to text  (base64) and put it inside a json property, but this added the overhead of encoding and later decoding server side, and the size of the payload that would increase significantly (ok for little images but not ok for potentially big files)

Reply:
ok, so why not rename data to metada?

Reply:
and file by attachement?

Reply:
that is fine for me..

Reply:
done